datadog_lambda/trigger.py

```diff
@@ -0,0 +1,182 @@
+# Unless explicitly stated otherwise all files in this repository are licensed
+# under the Apache License Version 2.0.
+# This product includes software developed at Datadog (https://www.datadoghq.com/).
+# Copyright 2019 Datadog, Inc.
+
+import base64
+import gzip
+import json
+from io import BytesIO, BufferedReader
+
+
+EVENT_SOURCES = {
+    "aws:dynamodb": "dynamodb",
+    "aws:kinesis": "kinesis",
+    "aws:s3": "s3",
+    "aws:sns": "sns",
+    "aws:sqs": "sqs",
+}
+
+
+def get_first_record(event):
+    records = event.get("Records")
+    if records and len(records) > 0:
+        return records[0]
+
+
+def read_event_source(event):
+    event_source = event.get("eventSource") or event.get("EventSource")
+    return EVENT_SOURCES.get(event_source)
+
+
+def get_event_source(event):
+    """
+    Attempts to determine the source of the trigger event
+    """
+    event_source = read_event_source(event)
+
+    request_context = event.get("requestContext")
+    if request_context and request_context.get("stage"):
+        event_source = "api-gateway"
+
+    if event.get("awslogs"):
+        event_source = "cloudwatch-logs"
+
+    event_detail = event.get("detail")
+    cw_event_categories = event_detail and event_detail.get("EventCategories")
+    if event.get("source") == "aws.events" or cw_event_categories:
+        event_source = "cloudwatch-events"
```

Comment:
AWS made this very confusing, just flagging in case this matters https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/WhatIsCloudWatchEvents.html